### PR TITLE
[Bugfix] Fix bias for 0-dim tensors in gemm

### PR DESCRIPTION
**Pull request by yaox12**
# Description

For `te_gemm` and `te_grouped_gemm`,
- If `grad == false`, we should do nothing to `bias`.
- If `grad == true`, we should calculate `grad_bias` here.

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refractor

# Checklist:

- [ ] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [ ] The functionality is complete
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: transformer_engine/pytorch/csrc/extensions/gemm.cu [hunk 1]
```diff
@@ -17,7 +17,8 @@ void te_gemm(at::Tensor A, at::Tensor A_scale_inverse, transformer_engine::DType
   using namespace transformer_engine;
   if (A.data_ptr() == nullptr || B.data_ptr() == nullptr) {
     if (D.data_ptr() != nullptr && !accumulate) D.zero_();
-    if (bias.data_ptr() != nullptr) bias.zero_();
+    // torch.sum is able to handle 0-dim tensors
+    if (bias.data_ptr() != nullptr && grad) bias.copy_(B.sum(0));
```
Comment (timmoon10) [inline transformer_engine/pytorch/csrc/extensions/gemm.cu:21]: We should handle the case where `B` has no data:
```suggestion
    if (bias.data_ptr() != nullptr && grad) {
      if (B.data_ptr() == nullptr) {
        bias.zero_();
      } else {
        bias.copy_(B.sum(0))
      }
    }
```
It seems we are checking `data_ptr() == nullptr` in order to check for empty tensors. As an alternative, it may be better to make our intention explicit:
```c++
  if (A.numel() == 0 || B.numel() == 0) {
    if (D.numel() != 0 && !accumulate) D.zero_();
    // torch.sum is able to handle 0-dim tensors
    if (bias.numel() != 0 && grad) bias.copy_(B.sum(0));
    if (pre_gelu_out.numel() != 0) pre_gelu_out.zero_();
    return;
  }
```

Comment (yaox12) [reply transformer_engine/pytorch/csrc/extensions/gemm.cu]: Updated.

**Review by ksivaman — APPROVED** [review]
LGTM, thanks!

**Comment by yaox12** [comment]
I'll merge this PR since the CI passes except for a doc test failing with irrelevant error.
